### PR TITLE
hotfix/Fix_UT_for_ConnectionHandler_in_Genivi

### PR DESCRIPTION
**Pull request by OHerasym**
Added ConnectionHandler unit tests.

Related:
        APPLINK-19073

### CODE REVIEW

**Comment by monstersem** [comment]
Please review: @VVeremjova , @Kozoriz, @iAndrew5 , @AGaliuzov

File: src/components/connection_handler/include/connection_handler/connection_handler_impl.h [hunk 1]
```diff
@@ -461,6 +461,7 @@ class ConnectionHandlerImpl : public ConnectionHandler,
 #ifdef BUILD_TESTS
   // Methods for test usage
  public:
+  DeviceMap &getDeviceList();
```
Comment (pestOO) [inline src/components/connection_handler/include/connection_handler/connection_handler_impl.h]: Could we return copy?
Or use const reference?

File: src/components/connection_handler/test/connection_handler_impl_test.cc [hunk 2]
```diff
@@ -401,51 +401,6 @@ TEST_F(ConnectionHandlerTest,OnFindNewApplicationsRequestWithoutSession) {
   connection_handler_->OnFindNewApplicationsRequest();
 }
 
-TEST_F(ConnectionHandlerTest, OnMalformedMessageCallback) {
```
Comment (pestOO) [inline src/components/connection_handler/test/connection_handler_impl_test.cc]: We has no such logic?

Comment (anosach-luxoft) [inline src/components/connection_handler/test/connection_handler_impl_test.cc]: +1. Why we remove this tests?

Comment (iAndrew5) [inline src/components/connection_handler/test/connection_handler_impl_test.cc]: +1

**Comment by iAndrew5** [comment]
Reviewed

**Comment by Kozoriz** [comment]
reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Comment by Kozoriz** [comment]
Reviewed

**Comment by pestOO** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
@OHerasym, wrong branch name. Should be `hotfix`. And rebase, please

**Comment by OHerasym** [comment]
@anosach-luxoft , Done

**Pull request merged**